### DISCUSSION TITLE
return results with a different search engine like https://duckduckgo.com/

### DISCUSSION THREAD

**Discussion by jim-perkins** [Kaljurand/K6nele#88] [Q&A] [answered]
Hello
I have K6nele working pretty well for me with the Kaldi Gstreamer and the tedlium_nnet_ms_sp_online model.  thanks so much.
When I click on the K6nele icon and speak the results are returned by www.google.com .  Is it possible to modify the domain so that the results are returned by https://duckduckgo.com/  ?

I think I have an idea where this occurs but have no idea if it can be changed in the UI or with a simple patch.  I could compile my own version of K6nele but have no idea where to start or if this is possible.  thanks

https://developer.android.com/reference/android/content/Intent#ACTION_WEB_SEARCH

Input: getStringExtra(SearchManager.QUERY) is the text to search for. If it is a url starts with http or https, the site will be opened. If it is plain text, Google search will be applied.

https://github.com/Kaljurand/K6nele/blob/b7febb8334506f2ee81a90994c239e84e8766110/app/src/main/java/ee/ioc/phon/android/speak/activity/AbstractRecognizerIntentActivity.java#L310

**Comment by Kaljurand** [comment, marked as answer]
Yes, by default ``ACTION_WEB_SEARCH`` is launched. This happens here:

https://github.com/Kaljurand/speechutils/blob/aa8d69d5aaf4041e7d09948cbbafd9c32d348f48/app/src/main/java/ee/ioc/phon/android/speechutils/utils/IntentUtils.java#L71

But any app can provide an activity to service this intent, e.g. if you have the DuckDuckGo app (com.duckduckgo.mobile.android) installed, you can configure it always to be used for ``ACTION_WEB_SEARCH``. In fact, it tries to convince you to do this during the first launch. After that the Kõnele output will be sent into the DuckDuckGo app. In general there is nothing in Kõnele that would lock it into using "www.google.com". (Also, it seems the bit about "_Google_ search will be applied" is incorrect in the Android documentation.)

Another solution is to set up a Kõnele rewrite rule. (Some examples are here: http://kaljurand.github.io/K6nele/docs/et/rewrites.html.)
This allows you to rewrite the recognition result into the ``VIEW``-intent, e.g. a rewrite rule table with a single row having these cell values will do:

- App: ``:``
- Utterance: ``(.+)``
- Command: ``activity``
- Arg1: ``{"action": "android.intent.action.VIEW", "data": "https://duckduckgo.com/?q=$1"}``

If you click on <http://tinyurl.com/ivkfjcvy> (in your Android device, where Kõnele is installed) you should be able to import the above table into Kõnele. Give it any name and enable it in the rewrites list. From then on all recognition results issued in the Kõnele launcher will be converted into the duckduckgo query URL, which is then opened in your default browser.

So hopefully there is no need to compile your own custom Kõnele for this use-case.

**Reply by jim-perkins** [reply]
> If you click on http://tinyurl.com/ivkfjcvy (in your Android device, where Kõnele is installed) you should be able to import the above table into Kõnele

This worked perfectly.   Thanks so much
